src/bundle-manifest.ts

```diff
@@ -201,16 +201,24 @@ export function generateSharedDepsMergeStrategy(minEntrypoints?: number):
  * bundle for an application shell.
  */
 export function generateShellMergeStrategy(
-    shell: UrlString, minEntrypoints: number): BundleStrategy {
+    shell: UrlString, minEntrypoints?: number): BundleStrategy {
   return (bundles: Bundle[]): Bundle[] => {
+    if (minEntrypoints === undefined) {
+      minEntrypoints = 2;
+    }
     const newBundles = generateSharedDepsMergeStrategy(minEntrypoints)(bundles);
-    const shellBundle = newBundles.find((bundle) => bundle.files.has(shell));
-    const sharedBundle =
-        newBundles.find((bundle) => bundle.entrypoints.size > 1);
-    if (shellBundle && sharedBundle && shellBundle !== sharedBundle) {
+    let shellBundle = newBundles.find(bundle => bundle.files.has(shell));
+    if (!shellBundle) {
+      throw new Error(`No bundle found containing specified shell ${shell}`);
+    }
+    const sharedBundles = newBundles.filter(
+        (bundle) => bundle.entrypoints.size >= minEntrypoints &&
+            bundle !== shellBundle);
+    for (const sharedBundle of sharedBundles) {
       newBundles.splice(newBundles.indexOf(shellBundle), 1);
       newBundles.splice(newBundles.indexOf(sharedBundle), 1);
-      newBundles.push(mergeBundles([shellBundle, sharedBundle]));
+      shellBundle = mergeBundles([shellBundle, sharedBundle]);
```

Comment:
There are definitely test cases where one entrypoint includes an html import to another entrypoint.  I can't recall if those tests were in bundler or build off-hand, but that scenario is not too far-fetched given how creative developers can be.

Hmm, but your point about the loop being misleading... I actually wrote the loop in the shell strategy code with the expectation that there might be scenarios where `generateSharedDepsMergeStrategy` should generate more than one shared bundle.  After testing it though, there don't appear to be situations where it does generate more than one.  it would I think there might be cases for it doing so in situations where there's a clear split in the dependency graph, but now I'm speculating...